Intrepid2: clang warnings about autological overlap compare

## Bug Report
@trilinos/Intrepid2

### Description
clang-10.0 emits warnings:

`.../Intrepid2_CubatureTensorDef.hpp:136:38: warning: overlapping comparisons always evaluate to true [-Wtautological-overlap-compare]`

### Steps to Reproduce
1. Build using a Clang toolchain with the Intrepid2 package enabled.

--- a/packages/intrepid2/src/Discretization/Integration/Intrepid2_CubatureTensorDef.hpp
+++ b/packages/intrepid2/src/Discretization/Integration/Intrepid2_CubatureTensorDef.hpp
@@ -133,7 +133,7 @@ namespace Intrepid2 {
         break;
       }
       default: {
-        INTREPID2_TEST_FOR_EXCEPTION( this->numCubatures_ != 2 || this->numCubatures_ != 3, std::runtime_error,
+        INTREPID2_TEST_FOR_EXCEPTION( !(this->numCubatures_ == 2 || this->numCubatures_ == 3), std::runtime_error,
                                       ">>> ERROR (CubatureTensor::getCubature): CubatureTensor supports only 2 or 3 component direct cubatures.");
       }
       }

